pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -497,4 +497,35 @@ public static AggregationFunctionType getAggregationFunctionType(String function
       }
     }
   }
+
+  /**
+   * Returns the stored {@code AggregationFunctionType} used to create the underlying value in the segment or index.
+   * Some aggregation function types share the same underlying value but are used for different purposes in queries.
+   * @param aggregationType the aggregation type used in a query
+   * @return the underlying value aggregation type used in storage e.g. StarTree index
+   */
+  public static AggregationFunctionType getAggregatedFunctionType(AggregationFunctionType aggregationType) {
```

Comment:
The stored type only applies to star-tree, so consider moving it to `AggregationFunctionColumnPair` and renaming to `getStoredType()`

Reply:
Addressed in https://github.com/apache/pinot/pull/12164/commits/2ca0cf588a484cf40c8ec57cb8c482b24f72ce71